app/services/exception_notifier_service.rb

```diff
@@ -0,0 +1,13 @@
+class ExceptionNotifierService < ApplicationService
+  def initialize(exception)
+    @exception = exception
+
+    super()
+  end
+
+  def call
+    raise @exception if Rails.env.development?
+
+    Sentry.capture_exception(e)
+  end
```

Comment:
Since `exception` is a runtime dependency, I think we don't need to pass it to the constructor. I know this is a common Ruby "idiom" but it's unnecessary IMO (also consumes a little bit more memory). Standard DI practice is to use constructors for "setup" dependencies and everything that's runtime should use arguments. In this case:

```ruby
class ExceptionNotifierService < ApplicationService
  def call(exception)
    # ...
  end
end
```
And then it could be injected into `AppointmentScheduler`:

```ruby
class AppointmentScheduler
  def initialize(exception_notifier)
    @exception_notifier = exception_notifier
  end

  def ...
    @exception_notifier.call(...)
  end
end
```

This allows us to test the interactions between the services without `allow_any_instance_of`.

Sorry if I'm noisy/redundant here, I read a book solely on dependency injection and now I'm brainwashed :joy:

Reply:
No worries, thank you for your input, this is the reason we have CRs :D